Code/Mantid/Framework/API/inc/MantidAPI/IRemoteJobManager.h

```diff
@@ -0,0 +1,277 @@
+API, etc.
+
+A typical sequence of calls when you use this interface would be:
+
+1) Authenticate/log-in (authenticate())
+2) Do transactions
+
+Where the sequence of calls within a transaction is:
+
+2.1) Start transaction (startRemoteTransaction())
+2.2) Do actions
+2.3) Stop transaction (stopRemoteTransaction())
+
+In 2.2, several types of actions are possible:
+- Submit a job to run on the (remote) compute resource (submitRemoteJob()).
+- Get status info for one or all jobs (queryRemoteJob() and
+queryAllRemoteJobs()).
+- Cancel a job (abortRemoteJob()).
+- Get list of available files for a transaction on the compute resource
+(queryRemoteFile())
+- Upload / download files ( uploadRemoteFile() and downloadRemoteFile()).
+
+
+Copyright &copy; 2015 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>.
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class MANTID_API_DLL IRemoteJobManager {
+public:
+  virtual ~IRemoteJobManager(){};
+
+  /**
+   * Status and general information about jobs running on (remote)
+   * compute resources.
+   */
+  struct RemoteJobInfo {
+    /// Job ID, usually assigned by a job scheduler as an integer
+    /// number or similar.
+    std::string id;
+    /// name of the job, whether given by the user or automatically
+    /// assigned by the job scheduler
+    std::string name;
+    /// Name of the script or executable. Depending on the specific
+    /// implementation, job scheduler, etc. this can be an
+    /// 'application' name, a script name or different ways of
+    /// specifying what is run
+    std::string runnableName;
+    /// Last status retrieved (typically: Pending, Running, Exited,
+    /// etc.). The values are implementation/job scheduler dependent.
+    std::string status;
+    /// ID of the transaction where this job is included
+    std::string transactionID;
+    /// Date-time of submission. No particular format can be assumed
+    std::string submitDate;
+    /// Date-time the job actually started running.  No particular
+    /// format can be assumed
+    std::string startDate;
+    /// Date-time the job finished. No particular format can be
+    /// assumed
+    std::string completionTime;
```

Comment:
Shouldn't the time-related have a more appropriate type (DateAndType for example)?